docs/reference/filters.md

```diff
@@ -1480,8 +1480,21 @@ of the http header and can be changed with an optional third
 parameter. If the third parameter is set skipper will use the
 defined HTTP header to put the request in the same client bucket,
 else the X-Forwarded-For Header will be used. You need to run skipper
-with command line flag `-enable-ratelimits`. Skipper will consume
-roughly 15 MB per filter for 100.000 clients.
+with command line flag `-enable-ratelimits`.
+
+One filter consumes memory calculated by the following formular, where
+N is the number of individual clients put into the same bucket, M the
+maximum number of requests allowed:
+
+```
+memory = N * M * 15 byte
```

Comment:
What is the method to estimate 15 bytes?

It seems more like
`memory = N * (135 + M * 34) ` 
It is `~ N * M * 34` for large M but for small M key size also matters:
```
# single bucket ratelimit, i.e. single ZSET
$ bin/skipper -inline-routes='* -> clusterRatelimit("group", 10, "1m") -> status(204) -> <shunt>' -enable-ratelimits -enable-swarm -swarm-redis-urls=localhost:6379

# repeat and query memory usage
$ curl -v localhost:9090

# memory usage
$ redis-cli
127.0.0.1:6379> keys *
1) "ratelimit.group.043a718774c572bd8a25adbeb1bfcd5c0256ae11cecf9f9c3f925d0e52beaf89"
127.0.0.1:6379> zcard ratelimit.group.043a718774c572bd8a25adbeb1bfcd5c0256ae11cecf9f9c3f925d0e52beaf89
(integer) 1
127.0.0.1:6379> memory usage ratelimit.group.043a718774c572bd8a25adbeb1bfcd5c0256ae11cecf9f9c3f925d0e52beaf89
(integer) 169
127.0.0.1:6379> zcard ratelimit.group.043a718774c572bd8a25adbeb1bfcd5c0256ae11cecf9f9c3f925d0e52beaf89
(integer) 2
127.0.0.1:6379> memory usage ratelimit.group.043a718774c572bd8a25adbeb1bfcd5c0256ae11cecf9f9c3f925d0e52beaf89
(integer) 203
127.0.0.1:6379> zcard ratelimit.group.043a718774c572bd8a25adbeb1bfcd5c0256ae11cecf9f9c3f925d0e52beaf89
(integer) 3
127.0.0.1:6379> memory usage ratelimit.group.043a718774c572bd8a25adbeb1bfcd5c0256ae11cecf9f9c3f925d0e52beaf89
(integer) 237
127.0.0.1:6379> zcard ratelimit.group.043a718774c572bd8a25adbeb1bfcd5c0256ae11cecf9f9c3f925d0e52beaf89
(integer) 4
127.0.0.1:6379> memory usage ratelimit.group.043a718774c572bd8a25adbeb1bfcd5c0256ae11cecf9f9c3f925d0e52beaf89
(integer) 271
127.0.0.1:6379> zcard ratelimit.group.043a718774c572bd8a25adbeb1bfcd5c0256ae11cecf9f9c3f925d0e52beaf89
(integer) 5
127.0.0.1:6379> memory usage ratelimit.group.043a718774c572bd8a25adbeb1bfcd5c0256ae11cecf9f9c3f925d0e52beaf89
(integer) 305
127.0.0.1:6379> zcard ratelimit.group.043a718774c572bd8a25adbeb1bfcd5c0256ae11cecf9f9c3f925d0e52beaf89
(integer) 6
127.0.0.1:6379> memory usage ratelimit.group.043a718774c572bd8a25adbeb1bfcd5c0256ae11cecf9f9c3f925d0e52beaf89
(integer) 339
...
```

Reply:
It is about local ratelimit :man_facepalming:

Reply:
no worries, thanks for double checking!